tools/cloud-build/images/cluster-toolkit-dockerfile/README.md

```diff
@@ -0,0 +1,59 @@
+
+## Build the Cluster Toolkit Docker Image
+To build the Cluster Toolkit Docker image, navigate to the directory the Dockerfile is present in and run the following command:
+
+```bash
+docker build --build-arg BASE_IMAGE=<base_image> \
+             --build-arg TERRAFORM_VERSION=<terraform_version> \
+             --build-arg PACKER_VERSION=<packer_version> \
+             --build-arg GO_VERSION=<go_version> \
+             --build-arg CLUSTER_TOOLKIT_VERSION=<cluster_toolkit_version> \
+             -t <image_name> .
+```
+
+Example:
+
+```bash
+docker build --build-arg CLUSTER_TOOLKIT_VERSION=v1.40.0 -t my-cluster-toolkit-image .
+```
+
+The above example builds an image tagged `my-cluster-toolkit-image` and sets the CLUSTER_TOOLKIT_VERSION to v1.40.0 while using the default values for other arguments.
+
+## Run the Cluster Toolkit Docker Image
+To run the Cluster Toolkit Docker image, use the following command:
+
+```bash
+docker run -v ~/.config/gcloud/:/root/.config/gcloud -v $(pwd):/data <image_name> <gcluster_command>
+```
+
+This command runs the Cluster Toolkit Docker image and allows the `gcluster` binary to access your Google Cloud credentials and local files. Here's a breakdown:
+
+* `-v ~/.config/gcloud/:/root/.config/gcloud`: This part mounts your local Google Cloud configuration directory `(~/.config/gcloud)` to the `/root/.config/gcloud` directory inside the container. This allows the `gcluster` binary to access your credentials and interact with Google Cloud resources.
+* `-v $(pwd):/data`: This part mounts your current working directory `($(pwd))` to the `/data` directory inside the container. This allows you to access any files in your current directory (like blueprint files) from within the container. You can then reference these files in your `<gcluster_command>` using the `/data` path.
+* `<image_name>`: Replace this with the name of your Docker image.
+* `<gcluster_command>`: Replace this with the `gcluster` command you want to execute. The Cluster Toolkit Docker image has `ENTRYPOINT ["gcluster"]` in its Dockerfile. This means that the `gcluster` command is automatically executed when the container starts, and any arguments provided after `<image_name>` in the docker run command are passed as arguments to `gcluster`.
+
+When using the docker container to run `gcluster` commands that either deploy or modify cloud resources, it's strongly recommended to save a local copy of the deployment folder that's generated inside of the docker container. This can be done by using the `--out` flag in combination with some `gcluster` commands to set the output directory where the `gcluster` deployment directory will be created. Ideally, the output directory should be a directory mounted from your local machine to the docker container. This ensures that the deployment folder persists even after the container exits, allowing you to access and manage the deployment artifacts even after the container is removed.
```

Comment:
I think asking users to set `--out` to a directory *inside* the container would make it confusing.

Can you see if there's a way to automatically set the output directory (i.e. `--out`) for relevant commands?

Maybe an easier alternative is to copy the `gcluster` binary into `/data` and not provide any `--out` argument at all.  If we do this though, be sure to document that the `--out` variable is not supported when using this Docker image.